relay/config/dynamic_config.go

```diff
@@ -0,0 +1,57 @@
+package config
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/go-yaml/yaml"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+// LoadDynamicConfig loads the dyanmic configuration for a bundle if
+// a) dynamic configuration is enabled and b) a config file exists for
+// the requested bundle
+func (c *Config) LoadDynamicConfig(bundle string) map[string]interface{} {
+	retval := make(map[string]interface{})
+	if c.DynamicConfigRoot == "" {
+		log.Debugf("Dynamic config disabled.")
+		return retval
+	}
+	if fullPath := locateConfigFile(c.DynamicConfigRoot, bundle); fullPath != "" {
+		buf, err := ioutil.ReadFile(fullPath)
+		if err != nil {
+			log.Errorf("Reading dynamic config for bundle %s failed: %s.", bundle, err)
+			return retval
+		}
+
+		err = yaml.Unmarshal(buf, &retval)
+		if err != nil {
+			log.Errorf("Parsing dynamic config for bundle %s failed: %s.", bundle, err)
+			return retval
+		}
+		for k := range retval {
```

Comment:
Yep.

Assume this code is in a file named `iter.go`.

```go
package main

import (
	"fmt"
)

func main() {
	data := map[string]string{
		"a": "1",
		"b": "2",
		"c": "3",
	}
	fmt.Printf("Just keys:\n")
	for k := range data {
		fmt.Printf("%s\n", k)
	}

	fmt.Printf("Keys and values:\n")
	for k, v := range data {
		fmt.Printf("%s: %s\n", k, v)
	}
}
```

Compiling and running it yields:

```shell
10:30:44 samedi.local:operable/iter/$ ./iter
Just keys:
a
b
c
Keys and values:
c: 3
a: 1
b: 2
```